### PR TITLE
Fixed noonce runtime autocorrection

### PR DESCRIPTION
**Pull request by xaionaro**
See: https://github.com/marinu666/PHP-btce-api/issues/6

### CODE REVIEW

File: btce-api.php [hunk 1]
```diff
@@ -95,10 +95,11 @@ public function apiQuery($method, $req = array()) {
         if(isset($result['error']) === true) {
             if(strpos($result['error'], 'nonce') > -1 && $this->RETRY_FLAG === false) {
                 $matches = array();
-                $k = preg_match('/:([0-9])+,/', $result['error'], $matches);
+                $k = preg_match('/:([0-9]+),/', $result['error'], $matches);
                 $this->RETRY_FLAG = true;
-                trigger_error("Nonce we sent ({$this->noonce}) is invalid, retrying request with server returned nonce: ({$matches[1]})!");
-                $this->noonce = $matches[1];
+                $new_noonce = $matches[1];
```
Comment (marinu666) [inline btce-api.php]: Fix looks good, just something I missed. Nicely done.

Can you change the lines below back that to not using the temporary variable new_noonce? And, just use matches[1] directly again. And, I'll get it merged.

Comment (marinu666) [inline btce-api.php]: oh also, file permissions have become 755, can you make this non-executable again. (0644 was the original)

**Comment by xaionaro** [comment]
Cleaned up.

However I disagree about `$new_noonce` :). Repeating `$matches[1]` is a bad coding style, IMHO.

**Comment by marinu666** [comment]
Two lines below we use it and it's used in context. Assigning an extra variable takes time, as speed is important for applications of the API, and PHP is pretty slow. The choice for this is just because we want it to be as fast as possible. I'll get this merged in :)

**Pull request merged**